Lib/test/test_concurrent_futures.py

```diff
@@ -1389,6 +1390,52 @@ def test_crash_big_data(self):
             with self.assertRaises(BrokenProcessPool):
                 list(executor.map(_crash_with_data, [data] * 10))
 
+    def test_gh105829_should_not_deadlock_if_wakeup_pipe_full(self):
+        # Issue #105829: The _ExecutorManagerThread wakeup pipe could
+        # fill up and block. See: https://github.com/python/cpython/issues/105829
+
+        # Lots of cargo culting while writing this test, apologies if
+        # something is really stupid...
+
+        self.executor.shutdown(wait=True)
+
+        if not hasattr(signal, 'alarm'):
+            raise unittest.SkipTest(
+                "Tested platform does not support the alarm signal")
+
+        def timeout(_signum, _frame):
+            raise RuntimeError("timed out while submitting jobs?")
+
+        thread_run = futures.process._ExecutorManagerThread.run
+        def mock_run(self):
+            # Delay thread startup so the wakeup pipe can fill up and block
+            time.sleep(5)
+            thread_run(self)
+
+        # Should be support.PIPE_MAX_SIZE but it is way too
+        # pessimistic here, would take too long. Assume 64k pipe
+        # buffer and add some margin...
+        job_num = 65536 * 2
```

Comment:
Please note that due to the large number of tasks, this tests takes quite a long time to run, about 60s on my old and tired machine. I don't know how to reliably trigger the bug with a shorter test while keeping the mocking to a minimum.

Reply:
I could maybe reduce this by a factor of 4. The wakeup code sends zero length byte objects, these will take 4 bytes to send (one integer describing the length zero followed by zero bytes of data). That would make it quite a lot faster.

I guess it would also be possible to estimate the required amount at the start by creating a pipe and counting how many times we can send ` b''` before it blocks, then add some margin.

Reply:
I'm surprised there's no way to control the size of the pipe, how could we do that?

Reply:
I added a fixup that reduces the pipe size on compatible platforms (Unix). This makes the test run in ~5s. On other platforms it falls back on a hardcoded 64k assumed pipe size but with the divide by 4, it still takes only ~12s.

Reply:
Does it still fail as frequently as it did before without the patch?

Reply:
The test has 100% failure rate on my system. I think there are only two ways that it could pass on a broken system: either we get the pipe buffer size wrong and start too few jobs to fill it, or the initial sleep in the manager thread is insufficient time to fill up the buffer. The first is very unlikely if the fcntl(F_SETPIPE_SZ) code runs (all unix platforms I think) since we get the true buffer size returned to us. On other platforms we just guess at 64k so here it can happen. The second failure case should also be unlikely, it takes only about 1s to submit 16k jobs (for the large 64k buffer fallback case). With a very loaded or slow system it could of course still fail but it should be rare.